R/IDateTime.R

```diff
@@ -13,29 +13,36 @@ as.IDate.default <- function(x, ..., tz = attr(x, "tzone")) {
 as.IDate.numeric <- function(x, origin = "1970-01-01", ...) {
   if (origin=="1970-01-01") {
     # standard epoch
-    structure(as.integer(x), class=c("IDate","Date"))
+    storage.mode(x) = 'integer'
```

Comment:
`storage.mode<-` also copies apparently:

```
y = pi:30*pi
tracemem(y)
# [1] "<0x7519040>"
storage.mode(y) = 'integer'
# tracemem[0x7519040 -> 0x2b85308]: 
```

And twice if `y` has shared assignment:

```
y = pi:30*pi
tracemem(y)
# [1] "<0x52705c0>"
z = y
storage.mode(y) = 'integer'
# tracemem[0x52705c0 -> 0x6444300]: 
# tracemem[0x6444300 -> 0x501dfa8]: 
```

Reply:
but we need at least one copy so storage mode of input is not updated, isn't it?

Reply:
@jangorecki right, but here I'm addressing @mattdowle's concern that we used `structure` in the first place since `class<-` is apt to make copies and hence can be slower than `structure(., class = )` approach.

Since `storage.mode<-` appears to aggressively copy, best not to use that (`as.integer` will only copy once regardless of `NAMED` status IIUC)